[Bug]: can't find project file in root directory

### Issue Description

When run in root directory such as `C:\`, MSBuild can't find project file

### Steps to Reproduce

1. Create project file `c:\hello.proj`
2. Change directory to `c:\`
3. Run `msbuild` 

### Expected Behavior

MSBuild runs default target in project `c:\hello.proj`

### Actual Behavior

MSBuild shows error:

> MSBUILD : error MSB1003: Specify a project or solution file. **The current working directory does not contain a project or solution file.**

### Versions & Configurations

MSBuild version 17.11.2+c078802d4 for .NET Framework
17.11.2.32701

No, we're not doing this on purpose. It works on .NET 9:

```sh-session
❯ ls *proj

    Directory: S:\

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           8/20/2024 11:08 AM              0 build.proj


❯ msbuild
MSBuild version 17.12.0-preview-24412-02+0716e1c2a for .NET Framework
MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.

❯ dotnet msbuild -tl:off
MSBuild version 17.12.0-preview-24374-02+48e81c6f1 for .NET
S:\build.proj : error MSB4025: The project file could not be loaded. Root element is missing.
```

This looks like a bug in .NET Framework's `System.IO.Directory.EnumerateFiles()`; check out the behavior of:

```C#
System.IO.Directory.SetCurrentDirectory(@"\");

var x = System.IO.Directory.GetFiles(".", "*.*proj");

System.Console.WriteLine(x.Length);
```

```sh-session
❯ .\play\io\bin\Debug\net472\io.exe
0

❯ .\play\io\bin\Debug\net6.0\io.exe
1
```

--- a/src/Build/Logging/ProfilerLogger.cs
+++ b/src/Build/Logging/ProfilerLogger.cs
@@ -301,7 +301,8 @@ namespace Microsoft.Build.Logging
                 DirectoryNotFoundException or
                 IOException or
                 UnauthorizedAccessException or
-                SecurityException)
+                SecurityException or
+                ArgumentException)
             {
                 Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("ErrorWritingProfilerReport", ex.Message));
             }

--- a/src/Build/Logging/ProfilerLogger.cs
+++ b/src/Build/Logging/ProfilerLogger.cs
@@ -297,19 +297,11 @@ namespace Microsoft.Build.Logging
 
                 Console.WriteLine(ResourceUtilities.GetResourceString("WritingProfilerReportDone"));
             }
-            catch (DirectoryNotFoundException ex)
-            {
-                Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("ErrorWritingProfilerReport", ex.Message));
-            }
-            catch (IOException ex)
-            {
-                Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("ErrorWritingProfilerReport", ex.Message));
-            }
-            catch (UnauthorizedAccessException ex)
-            {
-                Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("ErrorWritingProfilerReport", ex.Message));
-            }
-            catch (SecurityException ex)
+            catch (Exception ex) when (ex is
+                DirectoryNotFoundException or
+                IOException or
+                UnauthorizedAccessException or
+                SecurityException)
             {
                 Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("ErrorWritingProfilerReport", ex.Message));
             }

--- a/src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs
+++ b/src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs
@@ -1532,25 +1532,6 @@ namespace Microsoft.Build.UnitTests
         }
 
         /// <summary>
-        /// Verifies that when the /profileevaluation switch is used with invalid filenames an error is shown.
-        /// </summary>
-        [MemberData(nameof(GetInvalidFilenames))]
-        [WindowsFullFrameworkOnlyTheory(additionalMessage: ".NET Core 2.1+ no longer validates paths: https://github.com/dotnet/corefx/issues/27779#issuecomment-371253486.")]
-        public void ProcessProfileEvaluationInvalidFilename(string filename)
-        {
-            bool enableProfiler = false;
-            Should.Throw(
-                () => MSBuildApp.ProcessProfileEvaluationSwitch(new[] { filename }, new List<ILogger>(), out enableProfiler),
-                typeof(CommandLineSwitchException));
-        }
-
-        public static IEnumerable<object[]> GetInvalidFilenames()
-        {
-            yield return new object[] { $"a_file_with${Path.GetInvalidFileNameChars().First()}invalid_chars" };
-            yield return new object[] { $"C:\\a_path\\with{Path.GetInvalidPathChars().First()}invalid\\chars" };
-        }
-
-        /// <summary>
         /// Verifies that help messages are correctly formed with the right width and leading spaces.
         /// </summary>
         [Fact]

--- a/src/MSBuild/XMake.cs
+++ b/src/MSBuild/XMake.cs
@@ -43,6 +43,17 @@ using LoggerDescription = Microsoft.Build.Logging.LoggerDescription;
 using SimpleErrorLogger = Microsoft.Build.Logging.SimpleErrorLogger.SimpleErrorLogger;
 using TerminalLogger = Microsoft.Build.Logging.TerminalLogger.TerminalLogger;
 
+#if NETFRAMEWORK
+// Use I/O operations from Microsoft.IO.Redist which is generally higher perf
+// and also works around https://github.com/dotnet/msbuild/issues/10540.
+// Unnecessary on .NET 6+ because the perf improvements are in-box there.
+using Microsoft.IO;
+using Directory = Microsoft.IO.Directory;
+using File = Microsoft.IO.File;
+using FileInfo = Microsoft.IO.FileInfo;
+using Path = Microsoft.IO.Path;
+#endif
+
 #nullable disable
 
 namespace Microsoft.Build.CommandLine

